docs/scheduled-transactions/revised-spec.md

```diff
@@ -0,0 +1,157 @@
+# Scheduled transactions 
+
+An ordinary transaction must be submitted to the network with the signatures 
+of enough Ed25519 keys to activate all the Hedera keys required to sign it. 
+Otherwise it will resolve to `INVALID_SIGNATURE`. 
+
+Here we describe a new kind of _scheduled_ transaction that is not directly 
+submitted to the network, but rather created as part of a _schedule entity_ in 
+the network's action queue. An Ed25519 key "signs" a scheduled transaction
+by signing an ordinary transaction that either creates or affirms the 
+schedule entity.
+
+Along with a scheduled transaction, a schedule entity (or simply _schedule_) also contains,
+  1. An optional memo. 
+  2. An optional admin key that can be used to delete the schedule.
+  3. An optional account to be charged the service fee for the scheduled transaction.
+  4. A list of the Ed25519 keys that the network deems to have signed the scheduled transaction.
+  5. A flag that says if the scheduled transaction may be executed as soon as the schedule has collected enough signing keys. 
+
+The schedule entity type is managed by four new HAPI operations,
+  1. The `ScheduleCreate` transaction creates a new schedule (possibly with a non-empty list of signing keys).
+  2. The `ScheduleSign` transaction adds one or more Ed25519 keys to a schedule's list of affirmed signers.
+  3. The `ScheduleDelete` transaction removes a schedule from the network state.
+  4. The `GetScheduleInfo` query gets the current state of a schedule.
+
+It is important to understand that the bytes of a scheduled transaction are never 
+directly signed by an Ed25519 key. Only `ScheduleCreate` or `ScheduleSign` bytes
+are ever signed, in the ordinary way. 
+
+## The `ScheduleCreate` transaction
+  
+The `ScheduleCreateTransactionBody` message is to be,
+  
+```  
+message ScheduleCreateTransactionBody {  
+  SchedulableTransactionBody scheduledTransactionBody = 1; // The scheduled transaction; its TransactionID must not be set
+  string memo = 2; // An optional memo with a UTF-8 encoding of no more than 100 bytes
+  Key adminKey = 3; // An optional Hedera key which can be used to sign a `ScheduleDelete` and remove the schedule
+  bool executeImmediatelyOnValidSignatures = 4; // May the scheduled transaction be executed as soon as its schedule has enough signing keys? MUST be `true`.
+  AccountID scheduledPayerAccountID = 5; // The id of the account to be charged the service fee for the scheduled transaction at the consensus time that it executes (if ever).
+}  
+```  
+
+The new `SchedulableTransactionBody` message is a strict subset of the `TransactionBody` message which omits the
+top-level `TransactionID`, `nodeAccountID`, and `transactionValidDuration` fields; and does not allow the `ScheduleCreateTransactionBody`
+and `ScheduleSignTransactionBody` messages in its `data` element.
```

Comment:
What about omitting schedule delete?

Reply:
We can't support scheduling `ScheduleSign` or `ScheduleCreate` because of architectural limitations on how many triggered transactions can run per call to `handleTransaction`. (If you could schedule these, you could easily set up a "domino effect" that triggers multiple transactions.)

But `ScheduleDelete` can't trigger a transaction, so it's ok to schedule.

Reply:
Got it